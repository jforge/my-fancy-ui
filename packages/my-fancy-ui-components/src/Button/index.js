// @flow

import type {Node as NodeType} from 'react';

type PropsType = {
  children: NodeType,
  className?: string
};

import mergeClassNames from 'classcat';
import React from 'react';

const Button = (props: PropsType) => {
  const {className, ...rest} = props;
  const finalClassName = mergeClassNames([
    'mdl-button mdl-js-button mdl-button--raised mdl-js-ripple-effect mdl-button--accent',
    className
  ]);

  return <button {...rest} className={finalClassName} />;
};

export default Button;
