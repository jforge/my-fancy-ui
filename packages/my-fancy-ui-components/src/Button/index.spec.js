import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import Button from './index.js';

describe('<Button/>', () => {
  let props;

  beforeEach(() => {
    props = {
      foo: 'bar',
      children: 'Foo children'
    };
  });

  it('should render correctly.', () => {
    const wrapper = shallow(<Button {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should merge the classNames prop on the rendered node if given.', () => {
    const wrapper = shallow(<Button {...props} className="myFooClassName" />);

    expect(wrapper.prop('className')).toContain('myFooClassName');
  });
});
