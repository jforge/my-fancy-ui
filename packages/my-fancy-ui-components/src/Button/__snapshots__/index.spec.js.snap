// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> should render correctly. 1`] = `
<button
  className="mdl-button mdl-js-button mdl-button--raised mdl-js-ripple-effect mdl-button--accent"
  foo="bar"
>
  Foo children
</button>
`;
